src/spec-lang/tc/typecheck.ts

```diff
@@ -1569,50 +1408,62 @@ export function tcMemberAccess(expr: SMemberAccess, ctx: STypingCtx, typeEnv: Ty
     }
 
     if (
-        baseT instanceof SUserDefinedTypeNameType &&
-        baseT.definition instanceof ContractDefinition
+        baseT instanceof TypeNameType &&
+        baseT.type instanceof UserDefinedType &&
+        baseT.type.definition instanceof ContractDefinition
     ) {
         // First check if this is a type name
-        const type = resolveTypeDef([ctx[0], baseT.definition], expr.member);
```

Comment:
This if could also be simplified using resolveAny. This requires passing the version to `tc()` somehow.

Reply:
Again this should be done as follow-up cleanup, due to the effort of propagating version and context throughout the `tc()` code.